src/04-spin-boson-model.md

```diff
@@ -0,0 +1,776 @@
+    df.iloc[i,1:] = evals
+```
+
+```python
+df.plot(x="$\Delta E$",figsize=(10,8),ylim=[1.9,2],legend=True, 
+        title="Zoom in of the stationary states ($\omega=1$, $U=0.2$)     (Fig 3)");
+plt.ylabel("Energy");
+
+
+```
+
+<!-- #region -->
+
+
+The level splitting seen in Fig 3 is much smaller than those seen in Fig 2 at the primary resonance ($\Delta E \approx \omega$). We can therefore say that the effective coupling between levels is much less for the non-primary resonances.
+
+Fig 3 also shows us that the location of the resonance is somewhat shifted, i.e. the anti-crossing does not occur when $\Delta E = 3 \omega$ but instead $\Delta E \approx 3\omega$. This shift is known as the [Bloch-Siegert shift](https://en.wikipedia.org/wiki/Bloch-Siegert_shift) (see also [Cohen-Tannoudji](https://iopscience.iop.org/article/10.1088/0022-3700/6/8/007) and [Hagelstein](https://iopscience.iop.org/article/10.1088/0953-4075/41/3/035601)) and arises from the the effect of the interaction energy in the Hamiltonian ($E_{I}$). Specifically, the resonance condition should instead be written as $\Delta E + E_{I} = 3\omega$ and hence the value of $\Delta E$ needed for resonance is somewhat reduced. 
+
+We can see the splitting of the levels and shifting of the resonances more clearly by scanning through various values of the interaction strength $U$. Let's create an [animated gif](https://github.com/maxhumber/gif) to show this.
+<!-- #endregion -->
+
+```python
+# NOTE: THIS CELL TAKES ABOUT 2 MIN TO RUN
+
+dfs = []                        # This will hold the dataframes for each value of U
+Us = np.linspace(0, 0.6, 40)    # This will give us 40 frames in our animation from U=0 to U=0.6
+
+for U in Us:
+    
+    df = make_df_for_energy_scan("$\Delta E$", -4, 4, 201, two_state.shape[0])
+
+    for i, row in df.iterrows():
+        H =  row["$\Delta E$"]*two_state + 1*bosons + U*interaction
+        evals, ekets = H.eigenstates()
+        df.iloc[i,1:] = evals 
+        
+    dfs.append(df)
+```
+
+```python
+# Uses the gif library from https://github.com/maxhumber/gif
+@gif.frame
+def plot(df,j):
+    df.plot(x="$\Delta E$",figsize=(10,8),ylim=[-0.5,5.5],legend=True, 
+        title=f"Stationary states for $H = (\Delta E / 2) \sigma_z + \hbar\omega(a^{{\dagger}}a +1/2) + U( a^{{\dagger}} + a )\sigma_x$     ($\omega=1$, $U={Us[j]:.3f}$) ");
+    plt.ylabel("Energy");
+    
+    
+frames = []
+for j, df in enumerate(dfs):
+    frame = plot(df, j)
+    frames.append(frame)
+gif.save(frames, "./img/04-energy-levels.gif.png", duration=250)
+```
+
+```python
+Image(filename="./img/04-energy-levels.gif.png")
+```
+
+Fig 3 also suggests some new physics. A resonance at $\Delta E \approx 3\omega$ implies that it might be possible for the TSS transition from |+> to |-> to result in the emission of 3 smaller bosons instead of a single larger one (as we would normally expect) - so called "down conversion". We'll investigate this possibility shortly.
+
```

Comment:
EXCELLENT! At the beginning I thought it was wrong but now I got it. You got a nice intuition to plot the levels this way

Reply:
I actually saw it plotted this way in the Shirley paper (fig 1) https://journals.aps.org/pr/abstract/10.1103/PhysRev.138.B979 and I thought...oh...this is much better!